src/plugins/walletConnectToDapps/components/modal/callRequest/methods/SignTransactionConfirmation.tsx

```diff
@@ -0,0 +1,141 @@
+import { FaGasPump, FaWrench } from 'react-icons/fa'
+import { useTranslate } from 'react-polyglot'
+import { Card } from 'components/Card/Card'
+import { CircularProgress } from 'components/CircularProgress/CircularProgress'
+import { FoxIcon } from 'components/Icons/FoxIcon'
+import { Text } from 'components/Text'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+import type { ConfirmData } from '../CallRequestCommon'
+import { AddressSummaryCard } from './components/AddressSummaryCard'
+import { AmountCard } from './components/AmountCard'
+import { ContractInteractionBreakdown } from './components/ContractInteractionBreakdown'
+import { GasFeeEstimateLabel } from './components/GasFeeEstimateLabel'
+import { GasInput } from './components/GasInput'
+import { ModalCollapsableSection } from './components/ModalCollapsableSection'
+import { ModalSection } from './components/ModalSection'
+import { TransactionAdvancedParameters } from './components/TransactionAdvancedParameters'
+import { useCallRequestFees } from './hooks/useCallRequestFees'
+import { useIsInteractingWithContract } from './hooks/useIsInteractingWithContract'
+
+type CallRequest = WalletConnectEthSignTransactionCallRequest
+
+type Props = {
+  request: CallRequest['params'][number]
+  onConfirm(data: ConfirmData): void
+  onReject(): void
+}
+
+export const SignTransactionConfirmation = ({ request, onConfirm, onReject }: Props) => {
+  const walletConnect = useWalletConnect()
+  const { feeAsset } = useCallRequestFees(request)
+  const { isInteractingWithContract } = useIsInteractingWithContract({
+    evmChainId: walletConnect.evmChainId,
+    address: request.to,
+  })
+  const translate = useTranslate()
+  const cardBg = useColorModeValue('white', 'gray.850')
+  const {
+    state: { walletInfo },
+  } = useWallet()
+  const WalletIcon = walletInfo?.icon ?? FoxIcon
+
+  const form = useForm<ConfirmData>({
+    defaultValues: {
+      nonce: convertHexToNumber(request.nonce).toString(),
+      gasLimit: convertHexToNumber(request.gas).toString(),
+      speed: FeeDataKey.Average,
+    },
+  })
+
+  if (!walletConnect.connector || !walletConnect.dapp) return null
+  const address = walletConnect.connector.accounts[0]
+
+  if (isInteractingWithContract === null)
+    return (
+      <Center p={8}>
+        <CircularProgress />
+      </Center>
+    )
+
+  return (
+    <FormProvider {...form}>
+      <VStack p={6} spacing={6} alignItems='stretch'>
+        <ModalSection title='plugins.walletConnectToDapps.modal.sendTransaction.sendingFrom'>
+          <AddressSummaryCard address={address} icon={<WalletIcon w='full' h='full' />} />
+        </ModalSection>
+        <ModalSection
+          title={`plugins.walletConnectToDapps.modal.sendTransaction.${
+            isInteractingWithContract ? 'interactingWith' : 'sendingTo'
+          }`}
+        >
+          <AddressSummaryCard
+            address={request.to}
+            showWalletProviderName={false}
```

Comment:
Might want to keep it explicit or implicit everywhere since this has a default value, fine both ways